lib/Site.js

```diff
@@ -356,24 +363,39 @@ Site.prototype.generatePages = function () {
   // Render the final rendered page to the output folder.
   const { baseUrl } = this.siteConfig;
   const builtFiles = {};
-  const pages = this.siteConfig.pages || [];
   const processingFiles = [];
+  const pages = this.filePaths || [];
   this.pageModels = pages.map(page => this.createPageData({
     baseUrl,
-    pageSrc: page.src,
-    title: page.title,
+    pageSrc: page,
     pageTemplate: this.pageTemplate,
+    titlePrefix: this.siteConfig.prefix ? this.siteConfig.prefix + TITLE_PREFIX_SEPARATOR : '',
   }));
   this.pageModels.forEach((page) => {
     processingFiles.push(page.generate(builtFiles));
   });
   return new Promise((resolve, reject) => {
     Promise.all(processingFiles)
-      .then(resolve)
-      .catch(reject);
+      .then((frontMatterData) => {
+        this.frontMatterData = frontMatterData;
+        resolve();
+      })
+      .catch((err) => {
+        reject(err);
+      });
   });
 };
 
+/**
+ * Builds the index of files to be traversed as addressable pages
+ */
+Site.prototype.buildIndex = function () {
+  const addressableFiles = this.siteConfig.addressable;
+  const filePathIndex = walkSync(this.rootPath, { directories: false, ignore: ['_boilerplates'] })
+    .filter(currentPath => addressableFiles.some(fileName => currentPath.includes(fileName)));
```

Comment:
Should we check for the exact match in the file base name? (`index.md` should match but not `other-index.md`)
Also are we going to support/use glob pattern in the future e.g. (`**/index.md`)

Reply:
Good catch.

It's possible that we implement that in the future, perhaps using walkSync's glob support.